Resources/ProxyProvider/DIRACCAProxyProvider.py

```diff
@@ -347,3 +427,35 @@ def __createCertM2Crypto(self):
     userCertStr = userCert.as_pem()
     userPubKeyStr = userPubKey.as_pem(cipher=None, callback=util.no_passphrase_callback)
     return S_OK((userCertStr, userPubKeyStr))
+
+  def getFakeProxy(self, dn, time, group=None):
```

Comment:
Such a test method should not be in the real service

Reply:
As for this method, it really is not part of the tests, it is an additional helper method for creating a proxy without any substantial validation, and it can be used for a specific case(such as testing) where just need to generate a proxy with specific DN on the fly. Since this method uses the `DIRACCAProxyProvider` private methods and is additional, can do as follows, I'll rename this method with the underline prefix(for ex.: `_forceGenerateProxyForDN`) to avoid confusion and to show that it's for internal use, and add more clear describe.

Reply:
but this would be a valid proxy no ? 
Can you give a real life example of where this would be useful ?

Reply:
yes, it would be a valid proxy and in a real life is only used for my tests so far

Reply:
Hum, I still believe that this method should not be in defined in the Service, even more if it delivers a valid proxy.
If the purpose is test only, then it should not be in production code. This is very bad practice. 
if you really need private method, you can always hack your code to access them from outside, you can transform the private method in protected method and inherit the `DIRACProxyProvider` in a `TestDIRACProxyProvider`, etc.

But again, maybe there is a valid real life usecase which is not testing ? If you can think of one, please let me know

Reply:
hum, ok